dataprep/eda/basic/compute.py

```diff
@@ -1477,12 +1554,260 @@ def calc_stats(
+    n_zeros = stats_insights["Zeros"]
+    p_zeros = stats_insights["Zeros (%)"]
+    chisquare_value = stats_insights["Chi Square"]
+    skewness = stats_insights["Skewness"]
+    n_infinite = stats_insights["Infinite"]
+    p_infinite = stats_insights["Infinite (%)"]
+    n_negatives = stats_insights["Negatives"]
+    p_negatives = stats_insights["Negatives (%)"]
+
+    if float(p_missing) > 0.01:
+        insights.append(
+            (
+                ("Missing", column),
+                f"{column} has {n_missing}({p_missing}%) missing values",
+            )
+        )
+    if n_unique <= 1:
+        insights.append(
+            (("Constant", column), f"{column} has constant value {value_counts}")
+        )
+    if n_unique == value_counts:
+        insights.append(
+            (("Unique", column), f"{column} has unique values {value_counts}")
+        )
+    if chisquare_value > 0.999:
+        insights.append((("Uniform", column), f"{column} is uniformly distributed"))
+    if p_zeros > 0.05:
+        insights.append(
+            (
+                ("Zeros", column),
+                f"{column} has {n_zeros} ({round(p_zeros*100, 2)}%) zeros",
+            )
+        )
+    if skewness >= 20:
+        insights.append(
+            (
+                ("Skewed", column),
+                f"{column} is highly skewed (\u03B31 = {round(skewness, 4)})",
+            )
+        )
+    if p_infinite >= 0.01:
+        insights.append(
+            (
+                ("Infinite", column),
+                f"{column} has {n_infinite} ({round(p_infinite*100, 1)}) infinite values",
+            )
+        )
+    if p_negatives > 0.01:
+        insights.append(
+            (
+                ("Negatives", column),
+                f"{column} has {n_negatives} ({float(str(p_zeros*100)[:-2])}%) negatives",
+            )
+        )
+
+    return insights
+
+
+def _format_insights(
+    insights_all: List[Tuple[Tuple[str, str], str]]
+) -> Tuple[Dict[str, List[str]], Dict[str, List[str]]]:
+
+    overview_insights: Dict[str, List[str]] = {}
+    column_insight: Dict[str, List[str]] = {}
+    for insight_tuple in insights_all:
+        if (
+            insight_tuple[0][1] not in column_insight.keys()
+            and insight_tuple[0][1] != "Overview"
+        ):
+            column_insight[insight_tuple[0][1]] = [insight_tuple[1]]
+        elif (
+            insight_tuple[0][1] in column_insight.keys()
+            and insight_tuple[0][1] != "Overview"
+        ):
+            column_insight[insight_tuple[0][1]].append(insight_tuple[1])
+        if insight_tuple[0][0] not in overview_insights.keys():
+            overview_insights[insight_tuple[0][0]] = [insight_tuple[1]]
+        elif insight_tuple[0][0] in overview_insights.keys():
+            overview_insights[insight_tuple[0][0]].append(insight_tuple[1])
+    return overview_insights, column_insight
+
+
+def _new_format(olddc: Dict[str, List[str]]) -> Dict[int, List[Dict[str, str]]]:
```

Comment:
Please also make a comment about this function. Besides, as it is new format, is there a corresponding old format?

Reply:
This function is used to paginate insights for the overview section also set the orders for displaying.